spec/models/jwt_api_entreprise_spec.rb

```diff
@@ -24,6 +24,36 @@
     it { is_expected.to have_and_belong_to_many(:roles) }
   end
 
+  describe '.seven_days_ago_created_tokens' do
+    let!(:token) { create(:jwt_api_entreprise, iat: iat_value) }
+
+    subject(:model) { described_class }
+
+    context 'when the token was issued 6 days ago' do
+      let(:iat_value) { 6.days.ago.to_i }
+
+      its(:seven_days_ago_created_tokens) { is_expected.not_to be_exist token.id }
+    end
+
+    context 'when the token was issued 7 days ago' do
+      let(:iat_value) { 7.days.ago.to_i }
+
+      its(:seven_days_ago_created_tokens) { is_expected.not_to be_exist token.id }
+    end
+
+    context 'when the token was issued 8 days ago' do
+      let(:iat_value) { 8.days.ago.to_i }
+
+      its(:seven_days_ago_created_tokens) { is_expected.to be_exist token.id }
+    end
+
+    context 'when the token was issued 9 days ago' do
+      let(:iat_value) { 9.days.ago.to_i }
+
+      its(:seven_days_ago_created_tokens) { is_expected.not_to be_exist token.id }
+    end
+  end
```

Comment:
On peut faire plus lisible imo. 

Pour commencer, et pour pas tester tous les jours possibles je propose d'utiliser `Faker` pour générer des factories avec ces helpers : 

```ruby 
# Random date in the future (up to maximum of N days)
# Keyword arguments: days
Faker::Date.forward(days: 23) # => "Fri, 03 Oct 2014"

# Random date in the past (up to maximum of N days)
# Keyword arguments: days
Faker::Date.backward(days: 14) #=> "Fri, 19 Sep 2014"


describe '.seven_days_ago_created_tokens' do 
  subject { described_class.seven_days_ago_created_tokens } 

  let!(:newer_than_7_days_tokens) { create_list(:jwt_api_entreprise, :newer_than_7_days) }
  let!(:older_than_7_days_tokens)  { create_list(:jwt_api_entreprise, :older_than_7_days) }

  it 'returns tokens newer than 7 days' do 
    expect(subject).to contain_exactly(newer_than_7_days_tokens) 
  end 

  it 'does not return tokens older than 7 days' do 
    expect(subject).to_not include(*older_than_7_days_tokens) 
  end 
end 
```

Reply:
Juste une remarque : je trouve ça c'est très bien de déclarer explicitement le sujet des tests. Mais nous ne devrions pas confondre _subject_ et _actual_, soient respectivement l'objet qui va être testé et la valeur réelle obtenue suite au challenge qui lui est appliqué.

Reply:
Je ne suis pas certain de comprendre ce que tu veux dire, de mon côté il n'y a pas de confusion. On utilise simplement beaucoup `subject` et `its` lorsque ça simplifie la lecture des specs. S'il y a ambiguïté / confusion c'est levée en review au cas par cas.

Reply:
Ce que je veux dire, à partir de cet exemple qui décrit le comportement de la méthode de classe `seven_days_ago_created_tokens`, c'est qu'on peut distinguer ces 3 choses :
- le sujet (ici, c'est clairement la classe à tester, qui d'ailleurs correspond aussi à `described_class`)
- son challenge (c'est la méthode à appeler dessus, `seven_days_ago_created_tokens`)
- la valeur réelle, _actual_ (qui est retourné par le sujet confronté à son challenge)

Ça me fait préférer cette écriture :

```ruby
describe '.seven_days_ago_created_tokens' do 
  subject { described_class } 

  let!(:newer_than_7_days_tokens) { create_list(:jwt_api_entreprise, :newer_than_7_days) }
  let!(:older_than_7_days_tokens)  { create_list(:jwt_api_entreprise, :older_than_7_days) }

  it 'returns tokens newer than 7 days' do 
    expect(subject.seven_days_ago_created_tokens).to contain_exactly(newer_than_7_days_tokens) 
  end 

  it 'does not return tokens older than 7 days' do 
    expect(subject.seven_days_ago_created_tokens).not_to include(*older_than_7_days_tokens) 
  end 
end 
```

ou bien avec `its` :

```ruby
describe '.seven_days_ago_created_tokens' do
  subject { described_class }

  let!(:newer_than_7_days_tokens) { create_list(:jwt_api_entreprise, :newer_than_7_days) }
  let!(:older_than_7_days_tokens)  { create_list(:jwt_api_entreprise, :older_than_7_days) }

  its(:seven_days_ago_created_tokens) { is_expected.to contain_exactly(newer_than_7_days_tokens) }
  its(:seven_days_ago_created_tokens) { is_expected.not_to include(*older_than_7_days_tokens) }
end
```

L'avantage d'avoir ici utilisé la DSL d'RSpec en distinguant sémantiquement le sujet de la valeure réelle m'a permis de convertir directement les exemples `it` vers la syntaxe `its`.

Reply:
J'ai une préférence pour l'écriture où le subject est explicite, après en vrai je m'en fiche du moment que c'est consistant.